mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -769,10 +769,24 @@ pub enum NewClientError {
     InvalidEndpoint(#[from] EndpointError),
     /// Invalid AWS credentials
     #[error("invalid AWS credentials")]
-    ProviderFailure(#[from] mountpoint_s3_crt::common::error::Error),
+    ProviderFailure(#[source] mountpoint_s3_crt::common::error::Error),
     /// Invalid configuration
     #[error("invalid configuration: {0}")]
     InvalidConfiguration(String),
+    /// An internal error from within the AWS Common Runtime
+    #[error("Unknown CRT error")]
+    CrtError(#[source] mountpoint_s3_crt::common::error::Error),
+}
+
+impl NewClientError {
+    fn construction_error(e: mountpoint_s3_crt::common::error::Error) -> Self {
+        const AWS_ERROR_INVALID_ARGUMENT: i32 = 34;
+        if e.raw_error() == AWS_ERROR_INVALID_ARGUMENT {
```

Comment:
I'm a bit uncomfortable about using hard-coded number. Should we add a function `is_invalid_argument_error` in `mountpoint_s3_crt::common::error::Error` and use the enum from Rust binding directly?